trie/triedb/pathdb/history.go

```diff
@@ -520,14 +612,14 @@ func writeHistory(freezer *rawdb.ResettableFreezer, dl *diffLayer) error {
 	}
 	var (
 		start   = time.Now()
-		history = newHistory(dl.rootHash(), dl.parentLayer().rootHash(), dl.block, dl.states)
+		history = newHistory(dl.rootHash(), dl.parentLayer().rootHash(), dl.block, dl.states, dl.nodes)
```

Comment:
When the trienodebuffer of diskLayer is not nodebufferList but another type, dl.nodes data will still be written here, but this data will not be used by other types. I suggest that we should adhere to the principle of minimal writing and only write nodes data when using the nodebufferList type.

Reply:
It only keep the latest 90K blocks and won't occupy more disk space.